frontend/packages/console-app/src/components/modals/clone/clone-pvc-modal.tsx

```diff
@@ -70,74 +84,88 @@ const ClonePVCModal = withHandlePromise((props: ClonePVCModalProps) => {
           kind: PersistentVolumeClaimModel.kind,
           apiGroup: '',
         },
+        resources: {
+          requests: {
+            storage: `${requestedSize}${requestedUnit}`,
+          },
+        },
         accessModes: resource.spec.accessModes,
-        resources: resource.spec.resources,
       },
     };
 
-    return handlePromise(k8sCreate(PersistentVolumeClaimModel, pvcCloneObj)).then(close);
+    return handlePromise(k8sCreate(PersistentVolumeClaimModel, pvcCloneObj)).then(
+      (cloneResource) => {
+        close();
+        // redirected to the details page for persitent volume claim
+        history.push(resourceObjPath(cloneResource, referenceFor(cloneResource)));
+      },
+    );
   };
 
   return (
     <Form onSubmit={submit}>
       <div className="modal-content modal-content--no-inner-scroll">
         <ModalTitle>Clone</ModalTitle>
         <ModalBody>
-          <FormGroup label="Name" isRequired fieldId="ceph-clone-pvc-modal__name">
+          <FormGroup label="Name" isRequired fieldId="clone-pvc-modal__name">
             <TextInput
-              isRequired
               type="text"
-              name="ceph-clone-pvc-modal__name"
+              className="clone-pvc-modal--text-box-space"
               value={clonePVCName}
               onChange={setClonePVCName}
+              aria-label="Clone Pvc"
+            />
+          </FormGroup>
+          <FormGroup label="Size" isRequired fieldId="clone-pvc-modal__size">
+            <RequestSizeInput
+              name="requestSize"
+              onChange={requestedSizeInputChange}
+              defaultRequestSizeUnit={requestedUnit}
+              defaultRequestSizeValue={requestedSize}
+              dropdownUnits={dropdownUnits}
+              required
             />
           </FormGroup>
-          <div className="ceph-clone-pvc-modal__details">
+          <div className="clone-pvc-modal__details">
```

Comment:
```suggestion
          <div className="co-clone-pvc-modal__details">
```
if it's a generic let's prefix it by `co-`

Reply:
for all the instances.